packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -241,7 +241,7 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<TDate>(
 
       onRangePositionChange(nextSelection);
 
-      const isFullRangeSelected = rangePosition === 'end' && isRangeValid(utils, newRange);
+      const isFullRangeSelected = isRangeValid(utils, newRange);
```

Comment:
Are we sure that this was a wrong additional check? 🤔 
What problem are you trying to solve with this fix?
We clearly have quite a few tests failing with this change and quite a fundamental UX change: if there is value present—the picker will close when selecting the start date. I do feel that this change is somewhat unwarranted...
cc @flaviendelangle @noraleonte

Reply:
Yes for me the fix is not wanted.